### ISSUE TITLE
running aws/codebuild/windows-base locally?

### ISSUE THREAD

**Issue by ssippe** [aws/aws-codebuild-docker-images#328]
Hi, 

Is it possible to run aws/codebuild/windows-base locally? I can't find any documentation on it.

I would be helpful to know what tools are installed on what paths to aid in creating a working buildspec.yml.

Cheers

**Comment by subinataws** [comment]
@ssippe - Unfortunately, no. CodeBuild local is only supported on Linux_x86 today. We also don't have the Windows Dockerfile available on this GitHub repo yet.

**Comment by ssippe** [comment]
It would be really helpful if I could just look at the dockerfile to figure
out what resources are available. Can you show that?

**Comment by ssippe** [comment]
I'm not sure I can use CodeBuild for building my .net framework application. If the following improvements were made, it would work fine.

Suggested Improvements:
- make the .net framework curated docker images are available 
- make the .net framework curated docker files available so you could figure out how the environment is configured.
- Update the windows host version from the current 10.0.14393 so if you want to run your own docker image, you can the supported microsoft SDK containers. Anything that runs on 10.0.14393 is EOL.
- Make codebuild local is unavailable

Cheers

**Comment by adam-coster** [comment]
I would also love to have this feature.

**Comment by jyeagle** [comment]
Anything to make it easier to build legacy .Net Framework apps would be appreciated!  I am trying to migrate some builds to AWS codebuild and it would help to know what is available in the windows images.

**Comment by finnairbadalov** [comment]
Need this as well

**Comment by joebarbere** [comment]
This would REALLY be a time saver. Having instructions on how to setup docker to access the registry where the CodeBuild images are hosted would be great (I assume its just in a private AWS ECR registry at the moment). As @ssippe mentioned, even then Dockerfile would be helpful to understand what is included in the image.

**Comment by richarddowner** [comment]
We would also benefit from having access to the windows code build image or Dockerfile, as we would like to extend the image to contain additional required build resources.

**Comment by matthewko-acl** [comment]
I also agree with this, having more insight into what's in the codebuild image or docker file would be good.

**Comment by denniswed** [comment]
any updates on windows support? i am having a heck of a time getting codebuild local work right on windows, even with GIT bash.

**Comment by bestrus** [comment]
Something new for **_Windows_** @subinataws ? pls

**Comment by devt** [comment]
I have created a patch which fixes the issue for me using Git Bash. See [PR #619](https://github.com/aws/aws-codebuild-docker-images/pull/619/files) or [my version](https://github.com/devt/aws-codebuild-docker-images/blob/patch-1/local_builds/codebuild_build.sh)

Example execution which works for me is below
```bash
mkdir -p "c:/tmp/lcb"
./codebuild_build.sh -i my-codebuild:latest -a /mnt/c/tmp/lcb -c -m -s $(pwd) -b buildspec/test.yml
```
The strange thing here is that bash (at least this version bundled with GitBash) mangles the paths and does not allow you to submit a path like /mnt/c/something. It somehow injects "C:/Program Files/Git/" at the beginning.

The workaround is to invoke PowerShell with the same params. PowerShell is kind enough not to mangle, like bash does.

**Comment by jcrucesdeveloper** [comment]
Any updated on this? is it on main the official Windows supports for CodeBuild local?
